src/main/java/org/mule/extension/http/api/policy/HttpListenerPolicyParametersTransformer.java

```diff
@@ -91,6 +93,7 @@ public Message fromFailureResponseParametersToMessage(Map<String, Object> parame
       httpListenerResponseBuilder.setReasonPhrase(httpResponseAttributes.getReasonPhrase());
       return mapBuilder.build();
     } else {
+      httpListenerResponseBuilder.setBody(message.getPayload());
```

Comment:
Why just the body?

Reply:
The body is required to avoid a NPE at HttpResponseFactory.java:86. The other parameters are not needed to avoid it, and we don't want to propagate them.